sw/device/lib/crypto/drivers/kmac.h

```diff
@@ -39,11 +59,40 @@ typedef enum kmac_operation {
  */
 typedef enum kmac_error {
   kKmacOk = 0,
-  kKmacInternalError = 1,
-  kKmacArgsError = 2,
-  kKmacNotImplemented = 3,
+  kKmacInternalError,
+  kKmacNotIdle,
+  kKmacCfgWriteDisabled,
+  kKmacIntrErrPending,
+  kKmacCfgDisabledError,
+  kKmacFatalFaultError,
+  kKmacRecovFaultError,
+  kKmacArgsError,
+  kKmacDigestLenTooLongError,
+  kKmacUnsupportedKeySizeError,
+  kKmacNotImplemented,
+  kKmacCustomPaddingError,
+  kKmacUnsupportedPaddingLength,
```

Comment:
Responding to your question about error checking -- it's totally fine for the driver to have a different error granularity than the hardware does, so feel free to use whatever error codes make sense for the driver. It looks like that's what you've done already, so I think that's fine :slightly_smiling_face:

Reply:
I tried to make this struct as verbose as possible, it will probably get through few updates as I add more strict checks for error conditions.